python/lsst/pipe/base/pipelineIR.py

```diff
@@ -345,6 +418,36 @@ def _read_contracts(self, loaded_yaml):
             if isinstance(contract, str):
                 self.contracts.append(ContractIR(contract=contract))
 
+    def _read_labeled_subsets(self, loaded_yaml: dict):
+        """Process the subsets portion of the loaded yaml document
+
+        Parameters
+        ----------
+        loaded_yaml: `Mapping`
```

Comment:
As above, I think this needs to be `MutableMapping` (and you should probably state below that it's modified in-place).